cdc/scheduler/internal/tp/agent.go

```diff
@@ -27,31 +40,589 @@ type agent struct {
 	trans     transport
 	tableExec internal.TableExecutor
 
-	tables       map[model.TableID]*schedulepb.TableStatus
-	runningTasks map[model.TableID]*schedulepb.Message
+	// runningTasks track all in progress dispatch table task
+	runningTasks map[model.TableID]*dispatchTableTask
+
+	// pendingTasks is a queue of dispatch table task yet to be processed.
+	// the Deque stores *dispatchTableTask.
+	pendingTasks deque.Deque
```

Comment:
Why do we need a deque? can we handle tasks immediately? Owner control the total number of task that are running concurrently.